celery/backends/redis.py

```diff
@@ -139,13 +158,19 @@ def _consume_from(self, task_id):
         key = self._get_key_for_task(task_id)
         if key not in self.subscribed_to:
             self.subscribed_to.add(key)
-            self._pubsub.subscribe(key)
+            try:
+                self._pubsub.subscribe(key)
```

Comment:
Perhaps we could define a [context manager](https://docs.python.org/3.6/library/contextlib.html#contextlib.contextmanager) that handles the reconnection process.

Reply:
Done!